docker/Dockerfile.test-indy

```diff
@@ -16,4 +16,4 @@ RUN pip3 install --no-cache-dir \
 
 ADD --chown=indy:root . .
 
-ENTRYPOINT ["/bin/bash", "-c", "pytest \"$@\"", "--"]
+ENTRYPOINT black . --check --diff && pytest "$@" --
```

Comment:
Hm, does that work? Can you still run `test-indy aries_cloudagent.issuer` for example? I didn't think variables got replaced with this syntax

Reply:
Seems to work for me:

```
nbrempel@merlin ~/P/v/aries-cloudagent-python> ./scripts/run_tests -k "test_acquire_release_with_waiter"
Sending build context to Docker daemon     18MB
Step 1/7 : FROM ubuntu:18.04
 ---> 72300a873c2c
Step 2/7 : RUN apt-get update -y && 	apt-get install -y --no-install-recommends 	python3 	python3-pip 	python3-setuptools 	libsodium23 && 	rm -rf /var/lib/apt/lists/*
 ---> Using cache
 ---> 856e65452e63
Step 3/7 : WORKDIR /usr/src/app
 ---> Using cache
 ---> a993e8df2d34
Step 4/7 : ADD requirements.txt requirements.dev.txt ./
 ---> Using cache
 ---> 86da63903ecc
Step 5/7 : RUN pip3 install --no-cache-dir 	-r requirements.txt 	-r requirements.dev.txt
 ---> Using cache
 ---> 6bafa03bbfb6
Step 6/7 : ADD . .
 ---> Using cache
 ---> e830f0f97be2
Step 7/7 : ENTRYPOINT ["/bin/bash", "-c", "pytest \"$@\"", "--"]
 ---> Using cache
 ---> 7766619d29ca
Successfully built 7766619d29ca
Successfully tagged aries-cloudagent-test:latest
Skipping Indy-specific tests: python3-indy module not installed.
.                                                                                                                                                                                                    [100%]
------------------------------------------------------------------------- generated xml file: /usr/src/app/test-reports/junit.xml --------------------------------------------------------------------------

----------- coverage: platform linux, python 3.6.9-final-0 -----------
Name                                                                                       Stmts   Miss     Cover   Missing
---------------------------------------------------------------------------------------------------------------------------
aries_cloudagent/admin/server.py                                                             195    143    26.67%   54-57, 66, 76, 89-92, 97, 102-105, 132-147, 152-241, 251-259, 266-270, 288-292, 307-313, 328-331, 335, 340-404, 413, 419-420, 424-432
aries_cloudagent/cache/base.py                                                                79     11    86.08%   75, 103, 108-109, 131, 144-146, 150, 166, 168
aries_cloudagent/cache/basic.py                                                               28      6    78.57%   24-26, 67-68, 73
aries_cloudagent/commands/__init__.py                                                         20     15    25.00%   9, 18-29, 34-35
aries_cloudagent/commands/help.py                                                             20     15    25.00%   11-31, 35
aries_cloudagent/commands/provision.py                                                        34     20    41.18%   21, 28-39, 44-56, 60
aries_cloudagent/commands/start.py                                                            68     46    32.35%   26-27, 32-33, 38, 43-68, 74-104, 108
aries_cloudagent/config/argparse.py                                                          296    230    22.30%   49, 63-76, 87-126, 138-166, 177-299, 308-352, 363-397, 406-416, 427-450, 463-472, 483-500, 512-519, 530-586, 594-639, 650-694, 704-714, 725-779, 789-806
aries_cloudagent/config/base.py                                                               37     21    43.24%   40-43, 52-55, 64-67, 75-81, 97-98
aries_cloudagent/config/base_context.py                                                       10      3    70.00%   21, 29-30
aries_cloudagent/config/default_context.py                                                    57     31    45.61%   36-53, 58-141, 148-168
aries_cloudagent/config/injection_context.py                                                  63     35    44.44%   27-29, 34, 39, 44, 49, 54, 59, 63-64, 79-93, 101-106, 126, 130-133
aries_cloudagent/config/injector.py                                                           48     31    35.42%   17-19, 24, 29, 33, 39-43, 47-48, 52, 72-91, 95-97, 101
aries_cloudagent/config/ledger.py                                                             92     76    17.39%   24-30, 39-80, 86-163
aries_cloudagent/config/logging.py                                                           108     95    12.04%   25-35, 56-77, 104-232
aries_cloudagent/config/provider.py                                                           63     45    28.57%   16-18, 22, 33, 43-46, 50-67, 75-78, 82-84, 98-102, 106-113
aries_cloudagent/config/settings.py                                                           42     26    38.10%   17-19, 28-31, 40-44, 53-54, 62-63, 67, 71, 75, 79, 83, 87, 91, 95-97
aries_cloudagent/config/util.py                                                               37     27    27.03%   15-18, 26-27, 31-54, 58
aries_cloudagent/config/wallet.py                                                             44     37    15.91%   16-65
aries_cloudagent/connections/models/connection_record.py                                     131     53    59.54%   85-98, 103, 108, 140-148, 161-165, 177-178, 189-196, 206-211, 222-229, 237-242, 247, 252, 260-264
aries_cloudagent/connections/models/connection_target.py                                      23      6    73.91%   39-44
aries_cloudagent/connections/models/diddoc/diddoc.py                                          98     68    30.61%   59-61, 67, 82, 88, 94, 100, 116-121, 134, 158, 178-215, 232-305, 319, 324, 329
aries_cloudagent/connections/models/diddoc/publickey.py                                       58     23    60.34%   59-62, 68, 74, 80, 93, 129-134, 140, 146, 152, 158, 164, 173, 183, 188, 198
aries_cloudagent/connections/models/diddoc/service.py                                         41     21    48.78%   65-83, 89, 95, 101, 107, 113, 119, 125, 130-137
aries_cloudagent/connections/models/diddoc/util.py                                            32     26    18.75%   38, 53-60, 77-97, 111-114
aries_cloudagent/core/conductor.py                                                           158    124    21.52%   60-65, 70-138, 143-218, 222-231, 245-254, 263-267, 271-285, 301-309, 313, 330-345, 351, 365-370
aries_cloudagent/core/dispatcher.py                                                          114     77    32.46%   49-51, 55-57, 65, 71, 75-87, 109, 131-182, 209-228, 232, 256-260, 271-283, 292, 302-303
aries_cloudagent/core/error.py                                                                12      4    66.67%   11-13, 18
aries_cloudagent/core/plugin_registry.py                                                     139    116    16.55%   23, 28, 33, 38-115, 119-166, 176-181, 190-194, 203-210, 219-247, 251-273, 277
aries_cloudagent/core/protocol_registry.py                                                   101     80    20.79%   19-21, 26-32, 37, 42, 46-57, 61-69, 87-99, 115-116, 134-173, 179-197, 201
aries_cloudagent/holder/base.py                                                                7      1    85.71%   24
aries_cloudagent/holder/indy.py                                                              100     75    25.00%   36-37, 55-74, 102-128, 140-161, 183-232, 242-256, 266-286, 304-312, 334-345, 369-379
aries_cloudagent/indy/__init__.py                                                             12      7    41.67%   12-23, 28-29
aries_cloudagent/indy/error.py                                                                19     10    47.37%   15-16, 20, 24-25, 37-42
aries_cloudagent/indy/util.py                                                                  3      1    66.67%   8
aries_cloudagent/issuer/base.py                                                               11      1    90.91%   33
aries_cloudagent/issuer/indy.py                                                               77     55    28.57%   36-37, 43, 66-70, 76-78, 89-105, 130-142, 155-160, 187-235, 252-268, 285, 316-339
aries_cloudagent/ledger/base.py                                                               17      3    82.35%   23, 73-74
aries_cloudagent/ledger/indy.py                                                              421    357    15.20%   70-87, 96-114, 118-119, 124-131, 135-139, 143-149, 154-168, 178-180, 184-185, 205-261, 283-357, 367-379, 389-397, 411-438, 452-468, 491-593, 603-610, 621-651, 663-669, 677-684, 692-707, 716-733, 747-755, 759-762, 766-768, 772-792, 800, 807, 811-814, 820-838, 842-858, 862-880, 884-895, 909-922, 927-938, 948-959
aries_cloudagent/ledger/provider.py                                                           27     18    33.33%   21-43
aries_cloudagent/ledger/routes.py                                                            101     61    39.60%   90-108, 125-136, 153-164, 180-196, 212-227, 233
aries_cloudagent/messaging/agent_message.py                                                  174     97    44.25%   66-67, 75, 95, 106, 117, 128, 133, 138, 143, 156, 167, 188-197, 220-231, 244-247, 258, 268, 273-275, 284-288, 298, 309, 319, 328-331, 340-341, 356-362, 371-373, 410, 438-457, 471-472, 486-503, 517-523, 537-540
aries_cloudagent/messaging/base_handler.py                                                     9      1    88.89%   21
aries_cloudagent/messaging/credential_definitions/routes.py                                   60     60     0.00%   3-207
aries_cloudagent/messaging/decorators/attach_decorator.py                                    189     82    56.61%   50, 55, 89-91, 96, 142-145, 160-172, 198-201, 209-212, 250-251, 267, 273, 279-281, 287, 302-311, 317, 323, 329, 343-409, 422-437, 442-447, 462-467, 550-551
aries_cloudagent/messaging/decorators/base.py                                                 93     62    33.33%   32-38, 42, 46-48, 52, 56-57, 62, 69, 74, 78, 82, 86-88, 92-100, 110-130, 139-149, 153-154
aries_cloudagent/messaging/decorators/localization_decorator.py                               17      4    76.47%   34-37
aries_cloudagent/messaging/decorators/please_ack_decorator.py                                 16      3    81.25%   30-32
aries_cloudagent/messaging/decorators/signature_decorator.py                                  46     21    54.35%   46-49, 70-78, 93-95, 108-112, 116
aries_cloudagent/messaging/decorators/thread_decorator.py                                     33     10    69.70%   50-54, 65, 76, 86, 98, 110
aries_cloudagent/messaging/decorators/timing_decorator.py                                     26      7    73.08%   47-53
aries_cloudagent/messaging/decorators/trace_decorator.py                                      72     29    59.72%   53-61, 72, 83, 94, 105, 116, 127, 138, 149, 182-185, 196, 207, 219-221, 225-227
aries_cloudagent/messaging/decorators/transport_decorator.py                                  18      4    77.78%   37-40
aries_cloudagent/messaging/models/base.py                                                     94     54    42.55%   30-36, 51-61, 85, 100, 111, 125-130, 143-148, 164-169, 179, 189-195, 218, 233, 244, 259-267, 278, 289-290
aries_cloudagent/messaging/models/base_record.py                                             228    149    34.65%   31-37, 63-73, 83-88, 93, 98, 105, 110-113, 118, 127-128, 138-141, 151-155, 169-173, 183-187, 191, 204-219, 232-248, 266-280, 299-325, 342-348, 358-360, 366, 371, 383-391, 402-409, 414, 421-432, 436-438, 448-449, 453-459
aries_cloudagent/messaging/request_context.py                                                 59     26    55.93%   27-36, 47, 58, 63, 72, 83, 94, 105, 116, 127, 137, 148, 158, 168-174, 178
aries_cloudagent/messaging/responder.py                                                       39     21    46.15%   33-35, 50-58, 72-73, 95-103, 130-131, 135, 139, 143, 147
aries_cloudagent/messaging/schemas/routes.py                                                  61     61     0.00%   3-169
aries_cloudagent/messaging/util.py                                                            58     18    68.97%   45, 50-51, 71, 73, 98, 103, 124-136, 152
aries_cloudagent/protocols/actionmenu/definition.py                                            1      1     0.00%   3
aries_cloudagent/protocols/actionmenu/v1_0/base_service.py                                    12      4    66.67%   17, 23-27
aries_cloudagent/protocols/actionmenu/v1_0/controller.py                                       9      3    66.67%   19-21
aries_cloudagent/protocols/actionmenu/v1_0/driver_service.py                                  18      7    61.11%   28-35, 53-62, 66-68
aries_cloudagent/protocols/actionmenu/v1_0/handlers/menu_handler.py                           10      5    50.00%   24-32
aries_cloudagent/protocols/actionmenu/v1_0/handlers/menu_request_handler.py                   14      9    35.71%   24-38
aries_cloudagent/protocols/actionmenu/v1_0/handlers/perform_handler.py                        14      9    35.71%   24-41
aries_cloudagent/protocols/actionmenu/v1_0/messages/menu.py                                   23      4    82.61%   43-46
aries_cloudagent/protocols/actionmenu/v1_0/messages/menu_request.py                           13      1    92.31%   22
aries_cloudagent/protocols/actionmenu/v1_0/messages/perform.py                                19      3    84.21%   32-34
aries_cloudagent/protocols/actionmenu/v1_0/routes.py                                          94     58    38.30%   75-83, 95-100, 113-128, 140-155, 168-191, 197
aries_cloudagent/protocols/actionmenu/v1_0/util.py                                            26     19    26.92%   20-27, 35-56
aries_cloudagent/protocols/basicmessage/definition.py                                          1      1     0.00%   3
aries_cloudagent/protocols/basicmessage/v1_0/handlers/basicmessage_handler.py                 26     22    15.38%   23-61
aries_cloudagent/protocols/basicmessage/v1_0/messages/basicmessage.py                         26      7    73.08%   44-50
aries_cloudagent/protocols/basicmessage/v1_0/routes.py                                        25     13    48.00%   30-44, 50
aries_cloudagent/protocols/connections/definition.py                                           1      1     0.00%   3
aries_cloudagent/protocols/connections/v1_0/handlers/connection_invitation_handler.py          9      4    55.56%   25-33
aries_cloudagent/protocols/connections/v1_0/handlers/connection_request_handler.py            21     15    28.57%   26-46
aries_cloudagent/protocols/connections/v1_0/handlers/connection_response_handler.py           25     18    28.00%   28-58
aries_cloudagent/protocols/connections/v1_0/manager.py                                       418    365    12.68%   57-58, 69, 127-185, 207-250, 270-300, 316-422, 438-485, 511-560, 591-625, 651-680, 696-730, 746-776, 798-860, 868-872, 880-894, 903-905, 913-917, 925-930, 941-960, 972-1024, 1037-1061, 1074-1103, 1113-1125
aries_cloudagent/protocols/connections/v1_0/messages/connection_invitation.py                 50     17    66.00%   67-70, 84-89, 144-154
aries_cloudagent/protocols/connections/v1_0/messages/connection_request.py                    20      3    85.00%   42-44
aries_cloudagent/protocols/connections/v1_0/messages/connection_response.py                   18      2    88.89%   34-35
aries_cloudagent/protocols/connections/v1_0/models/connection_detail.py                       27      7    74.07%   24, 37, 57-59, 70, 81
aries_cloudagent/protocols/connections/v1_0/routes.py                                        161    101    37.27%   87-93, 167-189, 205-211, 251-274, 308-319, 352-364, 391-402, 415-427, 438-445, 462-486, 492
aries_cloudagent/protocols/discovery/definition.py                                             1      1     0.00%   3
aries_cloudagent/protocols/discovery/v1_0/handlers/disclose_handler.py                         7      3    57.14%   17-20
aries_cloudagent/protocols/discovery/v1_0/handlers/query_handler.py                           15      9    40.00%   19-28
aries_cloudagent/protocols/discovery/v1_0/routes.py                                           15      5    66.67%   45-49, 55
aries_cloudagent/protocols/introduction/definition.py                                          1      1     0.00%   3
aries_cloudagent/protocols/introduction/v0_1/base_service.py                                  12      4    66.67%   20, 26-30
aries_cloudagent/protocols/introduction/v0_1/demo_service.py                                  50     37    26.00%   43-76, 90-118
aries_cloudagent/protocols/introduction/v0_1/handlers/forward_invitation_handler.py           14      9    35.71%   21-38
aries_cloudagent/protocols/introduction/v0_1/handlers/invitation_handler.py                   13      8    38.46%   19-33
aries_cloudagent/protocols/introduction/v0_1/handlers/invitation_request_handler.py           17     11    35.29%   21-38
aries_cloudagent/protocols/introduction/v0_1/messages/forward_invitation.py                   19      3    84.21%   41-43
aries_cloudagent/protocols/introduction/v0_1/messages/invitation.py                           19      3    84.21%   39-41
aries_cloudagent/protocols/introduction/v0_1/messages/invitation_request.py                   18      3    83.33%   32-34
aries_cloudagent/protocols/introduction/v0_1/routes.py                                        20     12    40.00%   39-55, 61
aries_cloudagent/protocols/issue_credential/definition.py                                      1      1     0.00%   3
aries_cloudagent/protocols/issue_credential/v1_0/handlers/credential_ack_handler.py           15      9    40.00%   27-43
aries_cloudagent/protocols/issue_credential/v1_0/handlers/credential_issue_handler.py         19     13    31.58%   28-60
aries_cloudagent/protocols/issue_credential/v1_0/handlers/credential_offer_handler.py         19     13    31.58%   28-59
aries_cloudagent/protocols/issue_credential/v1_0/handlers/credential_proposal_handler.py      19     13    31.58%   28-61
aries_cloudagent/protocols/issue_credential/v1_0/handlers/credential_request_handler.py       22     15    31.82%   29-77
aries_cloudagent/protocols/issue_credential/v1_0/manager.py                                  264    222    15.91%   46-47, 58, 63-71, 92-107, 147-175, 186-209, 225-287, 297-345, 362-429, 442-459, 481-562, 574-592, 611-682, 692-708, 724-745, 754-772
aries_cloudagent/protocols/issue_credential/v1_0/messages/credential_issue.py                 25      2    92.00%   60, 65
aries_cloudagent/protocols/issue_credential/v1_0/messages/credential_offer.py                 28      2    92.86%   64, 69
aries_cloudagent/protocols/issue_credential/v1_0/messages/credential_request.py               25      2    92.00%   61, 66
aries_cloudagent/protocols/issue_credential/v1_0/messages/inner/credential_preview.py         48      9    81.25%   56, 61-67, 131, 142, 156
aries_cloudagent/protocols/issue_credential/v1_0/models/credential_exchange.py                93     34    63.44%   70-93, 98, 103, 135-144, 148
aries_cloudagent/protocols/issue_credential/v1_0/routes.py                                   320    224    30.00%   180-184, 200-209, 225-233, 257-315, 332-381, 404-481, 503-543, 559-602, 619-674, 691-740, 782-794, 810-814, 830-839, 854-883, 889
aries_cloudagent/protocols/present_proof/definition.py                                         1      1     0.00%   3
aries_cloudagent/protocols/present_proof/v1_0/handlers/presentation_ack_handler.py            15      9    40.00%   27-42
aries_cloudagent/protocols/present_proof/v1_0/handlers/presentation_handler.py                16     10    37.50%   27-50
aries_cloudagent/protocols/present_proof/v1_0/handlers/presentation_proposal_handler.py       19     13    31.58%   28-66
aries_cloudagent/protocols/present_proof/v1_0/handlers/presentation_request_handler.py        39     28    28.21%   33-116
aries_cloudagent/protocols/present_proof/v1_0/manager.py                                     197    165    16.24%   39-40, 51, 73-87, 97-111, 136-169, 186-199, 215-222, 265-440, 450-497, 513-585, 597-610, 623-639
aries_cloudagent/protocols/present_proof/v1_0/messages/inner/presentation_preview.py         130     53    59.23%   53-56, 140, 151-158, 163, 168, 181-193, 254, 270, 303-402, 407-410
aries_cloudagent/protocols/present_proof/v1_0/messages/presentation.py                        22      1    95.45%   61
aries_cloudagent/protocols/present_proof/v1_0/messages/presentation_ack.py                    13      1    92.31%   30
aries_cloudagent/protocols/present_proof/v1_0/messages/presentation_request.py                22      1    95.45%   64
aries_cloudagent/protocols/present_proof/v1_0/models/presentation_exchange.py                 61     16    73.77%   59-71, 76, 81, 100
aries_cloudagent/protocols/present_proof/v1_0/routes.py                                      261    163    37.55%   260-269, 285-293, 331-374, 391-443, 468-511, 531-586, 606-651, 668-722, 738-775, 787-797, 803
aries_cloudagent/protocols/present_proof/v1_0/util/indy.py                                    29     26    10.34%   25-102
aries_cloudagent/protocols/present_proof/v1_0/util/predicate.py                               20      7    65.00%   47-54, 67-69
aries_cloudagent/protocols/problem_report/definition.py                                        1      1     0.00%   3
aries_cloudagent/protocols/problem_report/v1_0/handler.py                                      8      4    50.00%   24-33
aries_cloudagent/protocols/problem_report/v1_0/message.py                                     39     13    66.67%   58-70
aries_cloudagent/protocols/routing/definition.py                                               1      1     0.00%   3
aries_cloudagent/protocols/routing/v1_0/handlers/forward_handler.py                           22     16    27.27%   21-45
aries_cloudagent/protocols/routing/v1_0/handlers/route_query_request_handler.py               14      8    42.86%   20-35
aries_cloudagent/protocols/routing/v1_0/handlers/route_query_response_handler.py               8      4    50.00%   18-24
aries_cloudagent/protocols/routing/v1_0/handlers/route_update_request_handler.py              14      8    42.86%   20-33
aries_cloudagent/protocols/routing/v1_0/handlers/route_update_response_handler.py             24     16    33.33%   24-57
aries_cloudagent/protocols/routing/v1_0/manager.py                                           101     77    23.76%   42-44, 55, 68-80, 102-125, 141-160, 164-166, 181-217, 227-232
aries_cloudagent/protocols/routing/v1_0/messages/forward.py                                   26      8    69.23%   34-38, 52-54
aries_cloudagent/protocols/routing/v1_0/messages/route_query_request.py                       19      3    84.21%   33-35
aries_cloudagent/protocols/routing/v1_0/messages/route_query_response.py                      21      3    85.71%   43-45
aries_cloudagent/protocols/routing/v1_0/messages/route_update_request.py                      18      2    88.89%   36-37
aries_cloudagent/protocols/routing/v1_0/messages/route_update_response.py                     18      2    88.89%   36-37
aries_cloudagent/protocols/routing/v1_0/models/paginate.py                                    14      3    78.57%   25-27
aries_cloudagent/protocols/routing/v1_0/models/paginated.py                                   18      5    72.22%   35-39
aries_cloudagent/protocols/routing/v1_0/models/route_query_result.py                          12      2    83.33%   24-25
aries_cloudagent/protocols/routing/v1_0/models/route_record.py                                20      6    70.00%   33-38
aries_cloudagent/protocols/routing/v1_0/models/route_update.py                                16      3    81.25%   28-30
aries_cloudagent/protocols/routing/v1_0/models/route_updated.py                               20      4    80.00%   38-41
aries_cloudagent/protocols/trustping/definition.py                                             1      1     0.00%   3
aries_cloudagent/protocols/trustping/v1_0/handlers/ping_handler.py                            18     13    27.78%   25-46
aries_cloudagent/protocols/trustping/v1_0/handlers/ping_response_handler.py                    9      5    44.44%   25-33
aries_cloudagent/protocols/trustping/v1_0/messages/ping.py                                    18      3    83.33%   34-36
aries_cloudagent/protocols/trustping/v1_0/messages/ping_response.py                           16      2    87.50%   30-31
aries_cloudagent/protocols/trustping/v1_0/routes.py                                           30     30     0.00%   3-60
aries_cloudagent/revocation/indy.py                                                           32     19    40.62%   18, 30-52, 64-67, 77, 83, 87-91
aries_cloudagent/revocation/models/indy.py                                                    21      3    85.71%   41-42, 46
aries_cloudagent/revocation/models/issuer_rev_reg_record.py                                  153     82    46.41%   87-103, 108, 113, 129-131, 135-172, 178-188, 192-208, 212-243, 254-258, 262-264, 268, 291-294, 305, 317-318, 322-323, 327
aries_cloudagent/revocation/models/revocation_registry.py                                    114     67    41.23%   40-49, 56-75, 80, 85, 90, 95, 100, 105, 110, 115, 120, 125, 130, 135, 140, 144-151, 155-156, 160-193, 197-200, 204-205
aries_cloudagent/revocation/routes.py                                                        115     69    40.00%   87-120, 162-171, 198-208, 230-240, 261-271, 291-305, 328-343, 348
aries_cloudagent/storage/base.py                                                              69     40    42.03%   112, 137-142, 147, 170, 181, 192, 203, 214, 224, 241-244, 248-253, 265-266, 270, 274, 278-288, 292
aries_cloudagent/storage/basic.py                                                            132    108    18.18%   28, 42-48, 68-72, 86-89, 103-106, 122-130, 143-145, 167, 177-203, 208-238, 263-267, 278, 294-310, 314-315, 319
aries_cloudagent/storage/indy.py                                                             108     80    25.93%   21-28, 42, 47, 57-66, 89-111, 131-139, 154-163, 176-182, 198-206, 228, 252-255, 266, 277, 293-310, 314-324, 330-332
aries_cloudagent/storage/provider.py                                                          15      7    53.33%   24-33
aries_cloudagent/storage/record.py                                                            10      5    50.00%   14-18
aries_cloudagent/transport/inbound/base.py                                                    22      7    68.18%   32-35, 40, 45, 64
aries_cloudagent/transport/inbound/delivery_queue.py                                          51     37    27.45%   26-27, 38, 55-56, 66-69, 82-91, 100-102, 111-114, 123-124, 133-135, 145-151
aries_cloudagent/transport/inbound/http.py                                                    57     45    21.05%   27-30, 34-40, 50-57, 64-66, 79-117, 130-136
aries_cloudagent/transport/inbound/manager.py                                                109     78    28.44%   39-48, 53-67, 79-94, 115, 125-127, 131, 135-136, 140-142, 163-179, 183-185, 193-201, 205-221, 230-233, 242-251
aries_cloudagent/transport/inbound/message.py                                                  9      5    44.44%   21-25
aries_cloudagent/transport/inbound/receipt.py                                                 83     34    59.04%   30-39, 50, 61, 72, 77, 88, 99, 110, 121, 132, 143, 154, 168, 179, 190, 201, 212, 223, 234, 245, 256, 267, 277-283
aries_cloudagent/transport/inbound/session.py                                                146     98    32.88%   24-25, 29, 52-73, 78, 83, 88, 92-95, 100, 105-113, 118, 123, 128, 133, 137-138, 142-143, 148, 157-163, 167-170, 179-181, 185-186, 195-212, 216-221, 235-240, 244-245, 249-250, 254-268, 272, 276
aries_cloudagent/transport/inbound/ws.py                                                      69     56    18.84%   28-31, 38, 42-44, 54-61, 68-70, 84-138
aries_cloudagent/transport/outbound/base.py                                                   32     10    68.75%   19-20, 25, 30, 34, 38-40, 53, 58
aries_cloudagent/transport/outbound/http.py                                                   35     23    34.29%   22-25, 29-38, 42-43, 56-67
aries_cloudagent/transport/outbound/manager.py                                               229    181    20.96%   50-60, 77-87, 91-95, 113-122, 140-164, 168-171, 175-176, 180-185, 189-196, 200-207, 212-224, 228, 238-254, 272-279, 288-293, 297-303, 310-389, 393-397, 401-405, 415-421, 425-430, 434-455, 459-461
aries_cloudagent/transport/outbound/message.py                                                18     13    27.78%   27-37, 47-48
aries_cloudagent/transport/outbound/ws.py                                                     21     10    52.38%   20-21, 25-26, 30-31, 45-49
aries_cloudagent/transport/pack_format.py                                                     90     70    22.22%   30-31, 52-101, 110-120, 148-155, 166-191
aries_cloudagent/transport/queue/base.py                                                      11      6    45.45%   55, 59-63
aries_cloudagent/transport/queue/basic.py                                                     48     36    25.00%   14-16, 20, 33-37, 51-77, 81, 85, 89, 93-95
aries_cloudagent/transport/stats.py                                                           38     26    31.58%   13-26, 30, 34, 38, 42, 46, 50, 54-58, 62
aries_cloudagent/utils/classloader.py                                                         75     56    25.33%   42-72, 97-120, 141-156, 162-175
aries_cloudagent/utils/http.py                                                                35     28    20.00%   43-58, 89-104
aries_cloudagent/utils/repeat.py                                                              52     33    36.54%   13-14, 18-21, 25-35, 40, 45, 49, 55, 63-65, 69, 73, 77-82, 86, 90
aries_cloudagent/utils/stats.py                                                              122     90    26.23%   14-17, 21-30, 34-52, 66-68, 73, 77-78, 82-86, 90, 94, 102-106, 110-115, 120, 125, 129-134, 138, 149-159, 164-170, 175-181, 189-197, 201, 206, 210
aries_cloudagent/utils/task_queue.py                                                         220    169    23.18%   13, 18-22, 27-34, 48-51, 55, 79-87, 91-94, 99, 104, 109-113, 117, 121, 138-149, 154, 159, 164, 173, 178, 183, 191, 195, 199-204, 208-212, 218-242, 251-254, 272-277, 298-309, 325-332, 342-364, 368-373, 377-381, 385-400, 404-409, 413, 417
aries_cloudagent/utils/temp.py                                                                 6      3    50.00%   10-12
aries_cloudagent/utils/tracing.py                                                            130    111    14.62%   43, 49-84, 90-107, 133-248
aries_cloudagent/verifier/base.py                                                              4      1    75.00%   17
aries_cloudagent/verifier/indy.py                                                             74     57    22.97%   35, 53-151, 174-198
aries_cloudagent/wallet/base.py                                                               44     28    36.36%   45, 139-145, 156-161, 173-190, 302
aries_cloudagent/wallet/basic.py                                                             117     84    28.21%   33-39, 44, 49, 54, 65, 69, 73, 92-103, 119-122, 136-138, 158-171, 184-185, 195-196, 212-214, 230-233, 247-249, 268-273, 291-297, 319-327, 347-355, 372-384
aries_cloudagent/wallet/crypto.py                                                            169    127    24.85%   60-63, 74, 88-91, 96-97, 111-122, 137-139, 154-158, 175-229, 302-310, 329-332, 350-363, 390-406, 419-436, 448-455, 469-497, 506-524
aries_cloudagent/wallet/indy.py                                                              250    190    24.00%   45-66, 71, 82, 87, 98, 109, 120, 131-139, 150-157, 171-182, 199-209, 224-267, 273-277, 297-314, 331-340, 354-356, 377-396, 406-417, 435-444, 465-469, 480-482, 501-509, 532-547, 568-579, 596-606, 610-619, 637-650, 657
aries_cloudagent/wallet/plugin.py                                                             30     22    26.67%   14-15, 22-44
aries_cloudagent/wallet/provider.py                                                           24     17    29.17%   22-41
aries_cloudagent/wallet/routes.py                                                            121     82    32.23%   57-58, 99-138, 154-159, 175-180, 202-223, 239-247, 263-276, 282
aries_cloudagent/wallet/util.py                                                               26     13    50.00%   9-10, 15, 20-22, 27, 42, 47-49, 54, 59
---------------------------------------------------------------------------------------------------------------------------
TOTAL                                                                                      11732   7239    38.30%

110 files skipped due to complete coverage.
Coverage XML written to file test-reports/coverage.xml

1 passed, 1496 deselected in 16.61 seconds
```

Reply:
Yeah but I mean.. through the manage script / dockerfile :)

Reply:
Can you give an example of a command that doesn't work? The manage script works for me as well and both bash scripts build and run the docker image so I think it should be the same.

Reply:
Hm, I just tried `./manage test-indy` and it bailed out early because black didn't like the format of a file in .vscode

Reply:
sounds like it's doing its job... 😝

the file isn't checked in, so it hasn't been checked before

Reply:
It just means I can't run the unit tests if I have any un-black-formatted files hanging around